src/sugar3/graphics/palettewindow.py

```diff
@@ -995,14 +995,14 @@ def notify_popdown(self):
         self._cursor_x = -1
         self._cursor_y = -1
 
-    def _ensure_palette_exists(self):
+    def force_palette(self):
```

Comment:
Why did you rename it force_palette?  That seems unnecessary, as ensure_palette_exsists is very descriptive.

Either way, if you're renaming a method and making it public, you should consider documenting it :)